src/ucs/datastruct/mpool.c

```diff
@@ -14,6 +14,12 @@
 #include <ucs/sys/sys.h>
 
 
+#ifdef SHM_HUGETLB
```

Comment:
could you put this definition into header file to eliminate duplication?
@yosefe is ucs/sys/sys.h good candidate?

Reply:
i don't think we need to define it at all
need to put the logic which uses it under `#ifdef` like it's already done in several places
because we really want to report that hugepages are not supported in such case
UNLESS - we define SHM_HUGETLB to 0 in ucs/sys/sys.h, AND whenever we use it, also handle a case where it's 0 (e.g not pass it to shmget if it's 0)

Reply:
So I do not see any further actionable item for me from your replies.  I do not see much use of defining the SHM_HUGETLB globally, because I think it is better to have new appearances of the flag to get errors from the compiler so that they can be examined.

Do you want anything else changed in these three trivial patches ?

Reply:
@kostikbel this patch is inconsistent.
From one hand, you are defining UCS_SHM_HUGETLB (in 2 places, which is also wrong, it should be one place)
From another hand, there is a bunch of `#ifdef SHM_HUGETLB ` in the code
need to decide on which approach to use, either define UCS_SHM_HUGETLB and check if it's 0 whenever it's used, or don't ever define it and put ifdef before using it